src/main/java/com/hubspot/jinjava/lib/fn/eager/EagerMacroFunction.java

```diff
@@ -124,24 +159,16 @@ public String reconstructImage() {
       // In case something not eager-supported encountered a deferred value
       result = macroFunction.reconstructImage();
     }
-    return prefix + result;
+    return prefix + result + suffix;
   }
 
-  private String getImportResourcePrefix(String prefix) {
+  private String getAliasedImportResourcePrefix(String importResourcePath) {
+    String prefix = "";
     String importAlias = fullName.split("\\.", 2)[0];
-    String importResourceKey;
-
     Object aliasMap = interpreter.getContext().get(importAlias);
     if (aliasMap instanceof DeferredValue) {
-      importResourceKey =
-        (String) (
-          (Map<String, Object>) ((DeferredValue) aliasMap).getOriginalValue()
-        ).get(Context.IMPORT_RESOURCE_PATH_KEY);
-      if (StringUtils.isEmpty(importResourceKey)) {
-        return "";
-      }
       PyMap deferredAliasMap = new PyMap(new HashMap<>());
-      deferredAliasMap.put(Context.IMPORT_RESOURCE_PATH_KEY, importResourceKey);
+      deferredAliasMap.put(Context.IMPORT_RESOURCE_PATH_KEY, importResourcePath);
```

Comment:
Could this cause problems if theres multiple imports?

Reply:
It shouldn't be an issue. The `deferredAliasMap` represents the whole object that is imported with an alias.
```
{% import 'file.html' as var %}
```
The `deferredAliasMap` will be put at `context.var`. A different import will have a separate map put onto the context with whatever key it's named with.

If there are multiple layers of imports, you would end up with something like:
```
context = {
  'var': {
    'import_resource_path': 'file.html',
    'nested': {
      'import_resource_path': 'nested.html',
      ... // other variables from nested.html
    },
    ... // other variables from file.html
  },
  ... // other variables on the context
}